File: backend/app/models/progression.py
from datetime import UTC, datetime

from pydantic import BaseModel, ConfigDict, Field


class StatsPublic(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    user_id: str
    chakra_xp: int
    vitality_xp: int
    focus_xp: int
    updated_at: datetime = Field(default_factory=lambda: datetime.now(UTC))


class AvatarStatePublic(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    user_id: str
    aura_level: int
    aura_label: str
    avatar_type: str
    updated_at: datetime = Field(default_factory=lambda: datetime.now(UTC))


class ProgressionPublic(BaseModel):
    stats: StatsPublic
    avatar: AvatarStatePublic

File: app/lib/models/progression.dart
class ProgressionSnapshot {
  const ProgressionSnapshot({
    required this.chakraXp,
    required this.vitalityXp,
    required this.focusXp,
    required this.auraLevel,
    required this.auraLabel,
    this.avatarType = 'neutral',
    this.streak = 0,
    this.totalXp = 0,
    this.nextLevelXp = 1000,
    this.weeklyCompletionRate = 0.0,
  });

  final int chakraXp;
  final int vitalityXp;
  final int focusXp;
  final int auraLevel;
  final String auraLabel;
  final String avatarType;
  final int streak;
  final int totalXp;
  final int nextLevelXp;
  final double weeklyCompletionRate;
}
